lib/reader.toit

```diff
@@ -124,6 +131,12 @@ class BufferedReader implements Reader:
   buffered -> int:
     return arrays_.size_in_bytes - first_array_position_
 
+  /**
+  The number of bytes that have been consumed from the BufferedReader.
+  */
+  cursor -> int:
```

Comment:
Maybe consumed is a better name given the description? I could see the name 'cursor' being used to delimit both the consumed part but also the buffered part.

Is it easy to add some tests of this functionality?